drainer.go

```diff
@@ -0,0 +1,49 @@
+package baseplate
+
+import (
+	"context"
+	"io"
+	"sync/atomic"
+)
+
+// Healthyer defines an interface to report healthy status.
+type Healthyer interface {
```

Comment:
Since this looks a little bit like a misspelling of healthier I wonder if we should have a different convention for interface names when the method is bound to a method of the form `Is[adjective]` especially when ending on -y to something like `type Health interface` instead.

Reply:
What about calling this `HealthChecker`?

Reply:
between the 2 I certainly like `HealthChecker` more

Reply:
Renamed to `HealthChecker` and `HealthCheckCloser`.